### DISCUSSION TITLE
Multiple request body examples not showing in Swagger after update to pydantic v2

### DISCUSSION THREAD

**Discussion by b-rad-c** [fastapi/fastapi#10095] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import Body, FastAPI
from pydantic import BaseModel

app = FastAPI()


class Item(BaseModel):
    name: str
    description: str | None = None
    price: float
    tax: float | None = None


@app.put("/items/{item_id}")
async def update_item(
    *,
    item_id: int,
    item: Annotated[
        Item,
        Body(
            examples=[
                {
                    "name": "Foo",
                    "description": "A very nice Item",
                    "price": 35.4,
                    "tax": 3.2,
                },
                {
                    "name": "Bar",
                    "price": "35.4",
                },
                {
                    "name": "Baz",
                    "price": "thirty five point four",
                },
            ],
        ),
    ],
):
    results = {"item_id": item_id, "item": item}
    return results
```

### Description

After updating to pydantic v2 and fast api 0.101.1 I am unable to get multiple request body examples to show in swagger. This is not a swagger problem but a problem with the way that fastapi generates the swagger page. The above example code was taken directly from the fast api documentation on [this page](https://fastapi.tiangolo.com/tutorial/schema-extra-example/#body-with-multiple-examples) under the section **_Body with multiple examples_** using the annotated python 3.10+ example.

**What I'm expecting**
To see a dropdown in the request path to show multiple request body examples.

This is the example screen shot from the documentation, I added a red rectangle to show the dropdown:
![fast api docs](https://github.com/tiangolo/fastapi/assets/25362581/79b0a323-a163-4693-8587-3dba73a29b95)

**What I'm actually seeing** 
The first example from the list in the code, but no dropdown to show the others.

Screenshot from my browser, notice there is no dropdown:
<img width="1548" alt="image" src="https://github.com/tiangolo/fastapi/assets/25362581/dec58129-c8b8-44fb-8006-a9ff1489f244">

**Steps:**
- copy the example code and paste into a file called main.py
- create a venv using python 3.10+ 
- activate the venv and run `pip install fastapi[all] pydantic`
- start the dev server: `uvicorn main:app --host 127.0.0.1 --port 1234`
- navigate in browser to: http://127.0.0.1:1234/docs#/default/update_item_items__item_id__put
- there is no dropdown allowing you to select different request body examples

It's worth noting that you can see the additional examples if you navigate to the route in the openapi json:

<img width="567" alt="image" src="https://github.com/tiangolo/fastapi/assets/25362581/eb65424b-48d6-480d-ab46-2f3c485b35a4">

### Operating System

macOS

### Operating System Details

11.7.7

### FastAPI Version

0.101.1

### Pydantic Version

2.1.1

### Python Version

3.10.6

### Additional Context

(venv) sh-3.2$ python -m pip freeze
annotated-types==0.5.0
anyio==3.7.1
certifi==2023.7.22
click==8.1.6
dnspython==2.4.2
email-validator==2.0.0.post2
exceptiongroup==1.1.3
fastapi==0.101.1
h11==0.14.0
httpcore==0.17.3
httptools==0.6.0
httpx==0.24.1
idna==3.4
itsdangerous==2.1.2
Jinja2==3.1.2
MarkupSafe==2.1.3
orjson==3.9.5
pydantic==2.1.1
pydantic-extra-types==2.0.0
pydantic-settings==2.0.3
pydantic_core==2.4.0
python-dotenv==1.0.0
python-multipart==0.0.6
PyYAML==6.0.1
sniffio==1.3.0
starlette==0.27.0
typing_extensions==4.7.1
ujson==5.8.0
uvicorn==0.23.2
uvloop==0.17.0
watchfiles==0.19.0
websockets==11.0.3

**Comment by jgould22** [comment, marked as answer]
You can read more about it here https://github.com/tiangolo/fastapi/discussions/9794

**Reply by b-rad-c** [reply]
Thanks, sorry I missed the duplicate question.

**Reply by tiangolo** [reply]
There's a new parameter `openapi_examples` just for this, to allow declaring examples that show up in Swagger UI while it implements support for multiple JSON Schema examples. It's available now in FastAPI 0.103.0, the new docs are here: https://fastapi.tiangolo.com/tutorial/schema-extra-example/#openapi-specific-examples 🎉

**Comment by austinorr** [comment]
@b-rad-c @jgould22 I opened a PR #9866 to fix this a month ago, but it's thus far failed to attract enough attention to get reviewed by a maintainer. If you'd like to pull it down, see if it's solving your problem, and help me review it, maybe there's a shot at getting this fixed if we can get more support.

**Reply by b-rad-c** [reply]
Thanks, I'll take a look, I think it's an important feature.
